source/Missions/Services/Agents/Handlers/WeaponPickupHandler.cs

```diff
@@ -0,0 +1,64 @@
+﻿using Common.Messaging;
+using Common.Network;
+using HarmonyLib;
+using Missions.Services.Agents.Messages;
+using Missions.Services.Network;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using TaleWorlds.Core;
+using TaleWorlds.MountAndBlade;
+
+namespace Missions.Services.Agents.Handlers
+{
+    public class WeaponPickupHandler
```

Comment:
This requires commenting also create a test case for each function. Create a new test file for this class in a similar directory structure